Fix prem

sannecottaar opened the pull request:
I cleaned out all the input files for prem, and fixed the values around the discontinuities in the mantle transition zone. I also tried to generalize SeismicRadiusTable, so it can handle tables with increasing depth or increasing radius (guess we could change the name SeismicRadiusTable then). If you guys are good with the file prem.txt, I will do the same for ak135 and iasp91.

ian-r-rose commented on burnman/data/input_seismic/PREMQL6.webarchive:
```diff
@@ -1,509 +0,0 @@
-bplist00�_WebMainResource�	
```
What is this file?  Is it necessary?

ian-r-rose commented:
I'm confused about all the different types of PREM.  Do they need to all be there, or can we merge them into one?

ian-r-rose commented:
Oops, egg on my face.  So much for reading comprehension... apologies, Sanne!

sannecottaar commented:
I could be a bit more elaborate. 
-- I replaced PRMQL6.webarchive, prem_lowermantle.txt, prem_table.txt, prem_table_nonSI.txt, prem_withQ.txt, grav_for_prem.txt with one file, prem.txt, wich includes Q values and gravity. 
-- I extended the example to plot these Q and gravity values.
-- I also changed my mind on seismologists always thinking along the radius, and also thought we should be able to accommodate files that have increasing depth (which prem.txt now is). The other option of increasing radius is tested by ak135 in the example. 
-- I think Fast() and Slow() could be removed, since we don't even want to use them in our inversion. But that aside.

sannecottaar commented:
Also included the option of depth bounds for internal_depth_list. This was important for a number of our examples (now that prem returns the entire earth and not just the lower mantle)

ian-r-rose commented:
So, it looks like the prem you have added here differs non-negligibly from what was there previously, at least in pressure (see, for example, example_grid.py).

sannecottaar commented:
Yeah, you are right that the pressure seems off at the CMB. I will improve
my integration. I'm afraid to say that the old version isn't necessarily a
better reference, as there were some weird jumps at the discontinuities,
not sure what happened there. If they were interpolated from somewhere
else, the values in the lower mantle might be fine.

On Wed, Jan 7, 2015 at 8:58 PM, Ian Rose <notifications@github.com> wrote:

> So, it looks like the prem you have added here differs non-negligibly from
> what was there previously, at least in pressure (see, for example,
> example_grid.py).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geodynamics/burnman/pull/107#issuecomment-69081832>.
>

ian-r-rose commented:
What is the latest on this, Sanne?

I just compared the old and new PREM models, and it seems that they are basically identical except in pressure.  As far as I can tell, the old pressures (as in, current master branch) are what is actually in Dziewonski and Anderson (1981).  

Where is an integration needed?  The original tables report all the information we need, as far as I can tell.

ian-r-rose commented on burnman/seismic.py:
```diff
@@ -222,45 +280,59 @@ def v_s(self, depth):
-           raise ValueError, "Pressure outside range of PREM"
-        radius = np.interp(pressure, self.table_pressure[::-1], self.table_radius[::-1] )
-        return self.earth_radius - radius
+        if self.table_depth[0]:
```
Should we no longer have this error check?  According to the documentation for interp, it returns the value on the edge of the model points if you request a point outside of the interval.

sannecottaar replied:
Well, rebasing to master got me a new git trauma...

And test_partitioning.py is failing. I'm not sure why, but I have to some
other things now.

@Ian with regards to pressure: Yeah, I just reinterpolated pressure again.
I guess at the time I didn't really look at the mantle transition zone
being messed up (see attached figure) and the center of the Earth missing.
It still frustrates me that I cannot reproduce it at greater depths, even
when I use a more detailed table. And I currently don't have pressure for
ak135 and iasp91 (but I could look around).

@Ian with regards to error message: I think this error is mostly useful to
allude people to the fact they are using the wrong unit. Maybe it should
say that?

On Tue, Jan 13, 2015 at 11:35 PM, Ian Rose <notifications@github.com> wrote:

> In burnman/seismic.py
> <https://github.com/geodynamics/burnman/pull/107#discussion-diff-22905349>
> :
>
> >
> >      def density(self, depth):
> >
> >          return self._lookup(depth, self.table_density)
> >
> >      def depth(self, pressure):
> > -        if pressure > max(self.table_pressure) or pressure < min(self.table_pressure)  :
> > -           raise ValueError, "Pressure outside range of PREM"
> > -        radius = np.interp(pressure, self.table_pressure[::-1], self.table_radius[::-1] )
> > -        return self.earth_radius - radius
> > +        if self.table_depth[0]:
>
> Should we no longer have this error check? According to the documentation
> for interp, it returns the value on the edge of the model points if you
> request a point outside of the interval.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geodynamics/burnman/pull/107/files#r22905349>.
>

ian-r-rose replied:
Oh, sorry, I wasn't clear. I agree this is a useful error message. It just looks like you are removing it.

sannecottaar replied:
Oops...

On Wed, Jan 14, 2015 at 4:54 PM, Ian Rose <notifications@github.com> wrote:

> In burnman/seismic.py
> <https://github.com/geodynamics/burnman/pull/107#discussion-diff-22950520>
> :
>
> >
> >      def density(self, depth):
> >
> >          return self._lookup(depth, self.table_density)
> >
> >      def depth(self, pressure):
> > -        if pressure > max(self.table_pressure) or pressure < min(self.table_pressure)  :
> > -           raise ValueError, "Pressure outside range of PREM"
> > -        radius = np.interp(pressure, self.table_pressure[::-1], self.table_radius[::-1] )
> > -        return self.earth_radius - radius
> > +        if self.table_depth[0]:
>
> Oh, sorry, I wasn't clear. I agree this is a useful error message. It just
> looks like you are removing it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geodynamics/burnman/pull/107/files#r22950520>.
>

sannecottaar replied:
ok, the test_partitioning.py led to the depth routine in PREM being very
wrong (example_geotherms was broken too, but test.sh didn't catch that).
Otherwise test_partitioning looks wrong too, because it puts GPa into the
anderson geotherm instead of Pa. Not sure if the has anything to do with
you not reproducing it Bob.


On Wed, Jan 14, 2015 at 5:11 PM, Sanne Cottaar <sanne.cottaar@gmail.com>
wrote:

> Oops...
>
> On Wed, Jan 14, 2015 at 4:54 PM, Ian Rose <notifications@github.com>
> wrote:
>
>> In burnman/seismic.py
>> <https://github.com/geodynamics/burnman/pull/107#discussion-diff-22950520>
>> :
>>
>> >
>> >      def density(self, depth):
>> >
>> >          return self._lookup(depth, self.table_density)
>> >
>> >      def depth(self, pressure):
>> > -        if pressure > max(self.table_pressure) or pressure < min(self.table_pressure)  :
>> > -           raise ValueError, "Pressure outside range of PREM"
>> > -        radius = np.interp(pressure, self.table_pressure[::-1], self.table_radius[::-1] )
>> > -        return self.earth_radius - radius
>> > +        if self.table_depth[0]:
>>
>> Oh, sorry, I wasn't clear. I agree this is a useful error message. It
>> just looks like you are removing it.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/geodynamics/burnman/pull/107/files#r22950520>.
>>
>
>

sannecottaar commented:
I think this one is good to go. I took out some more bugs by testing examples individually. The test suite is working.

tjhei commented on burnman/data/input_seismic/prem.txt:
```diff
@@ -0,0 +1,95 @@
+#Depth(m)	Radius(m)	Pressure(Pa)	Gravity(m/s^2)	Density(g/m^3)	Vp(m/s)	Vs(m/s)	Qk	Qmu
```
can we have a comment with the source here?

tjhei replied:
also did you do any editing/computations from the source?

sannecottaar replied:
I've calculated gravity, because although it is in the original paper, it isn't anywhere in the PREM data tables that I have. 
And values have been reinterpolated, and I gave discontinuities a finite width of 1km instead of 0 km, because I think various interpolations didn't deal with having two values for the same depth.

bobmyhill replied:
I don't like giving discontinuities a finite width - PREM is no longer PREM in this case. Would it be better to have a line in the scripts that imposes a small dz (dr, dp) when z[i] = z[i+1]? Something like this, perhaps:
# Create interpolatable depths list
for i, depth in enumerate(depths):
    if depth <= depths[i-1]:
        depth += 1e-12

tjhei commented on burnman/seismic.py:
```diff
@@ -45,7 +45,7 @@ def evaluate_all_at(self, depth_list):
 
 
-    def internal_depth_list(self):
+    def internal_depth_list(self,mindepth=0.,maxdepth=6371.e3):
```
What do these two arguments do? I don't think those are a good idea.

Please document.

sannecottaar replied:
With this you can get the depth list for e.g. part of the PREM model instead of the entire earth.

tjhei commented on burnman/seismic.py:
```diff
@@ -155,21 +154,55 @@ def K(self, depth):
+        Returns
+        -------
+        Qk : float or array of floats
+        Quality factor for bulk modulus at given depth(s).
```
does it have units?

tjhei commented on burnman/seismic.py:
```diff
@@ -155,21 +154,55 @@ def K(self, depth):
+        Returns
+        -------
+        QG : float or array of floats
+        Quality factor for shear modulus at given depth(s).
```
does it have units?

tjhei commented on burnman/seismic.py:
```diff
@@ -193,27 +226,39 @@ class SeismicRadiusTable(Seismic1DModel):
+            return np.array([self.table_depth[x] for x in range(len(self.table_depth)) if self.table_depth[x]>mindepth and self.table_depth[x]<maxdepth]) 
+        else:
+            depth = self.earth_radius - self.table_radius[::-1]
+            return np.array([depth[x] for x in range(len(depth)) if depth[x]>mindepth and depth[x]<maxdepth])
```
instead of creating this whole array every time this function is called, can you put the formula earth_radius-radius inside the ```[ ... ]```?

tjhei commented on burnman/seismic.py:
```diff
@@ -229,38 +282,46 @@ def density(self, depth):
+        if not len(self.table_depth) == 0:
+            depth = np.interp(pressure, self.table_pressure, self.table_depth )
+        else:
+            depth = np.interp(pressure, self.table_pressure[::-1], self.earth_radius - self.table_radius[::-1])
```
please pass table_radius and then convert to depth after conversion.

sannecottaar replied:
Not sure what you want me to change here?

tjhei replied:
Not needed if you do everything in radius, but I was suggesting something like
```
   depth =  self.earth_radius - np.interp(pressure, self.table_pressure,self.table_radius)
```
so you don't create another temporary array

tjhei commented on burnman/seismic.py:
```diff
@@ -229,38 +282,46 @@ def density(self, depth):
+        if not len(self.table_radius) == 0:
+            radius = np.interp(pressure, self.table_pressure, self.table_radius )
+        else:
+            radius = np.interp(pressure, self.table_pressure[::-1], self.earth_radius - self.table_depth[::-1])
```
same here

tjhei commented:
What is the reason for supporting radius and depth based tables in SeismicRadiusTable? It would be easier to convert depth to radius once while loading in the file. This would be faster and you need a lot less code.

sannecottaar commented:
I guess I attempted to keep the table the same as the input either
increasing radius or increasing depth. Yes, we could also flip the entire
table if needed.

On Thu, Jan 15, 2015 at 2:53 PM, Timo Heister <notifications@github.com>
wrote:

> What is the reason for supporting radius and depth based tables in
> SeismicRadiusTable? It would be easier to convert depth to radius once
> while loading in the file. This would be faster and you need a lot less
> code.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geodynamics/burnman/pull/107#issuecomment-70096365>.
>

sannecottaar commented:
There needs to be some kind of __init__ for every seismic model then, in case people define their own model.

ian-r-rose commented:
I am still seeing differences of ~0.1-0.2 GPa in pressure, especially in the core:

![compare_with_old_datafile](https://cloud.githubusercontent.com/assets/5728311/5764533/99cfff1c-9ca7-11e4-90d2-39b70816ba72.png)

The x-axis is depth and the y axis is difference with the old datafile for prem.  The biggest changes are your fixes for the transition zone, clearly, but the wiggles in the rest of it seem odd to me.

Again, the old prem_table.txt appears to be a direct transcription from Dziewonski and Anderson.  Is there some update to the pressure profile that I'm unaware of?  To my eyes, the new one looks less smooth:
![compre_data](https://cloud.githubusercontent.com/assets/5728311/5765124/d9c017fc-9cab-11e4-8ad2-1ac4bc4157ba.png)

sannecottaar commented:
Okay, this should be looking a lot better now. The pressure in PREM is fixed. 
Other changes are:
- adding a couple more 1D models (example_seismic.py is changed to show all these models)
- some of the models do not give gravity and pressure. I put in a simple cumtrapz integration to get these. They differ a bit from PREM, and that might be because there aren't enough layers in these models to this. Anything more precise, with spline fitting, I haven't gotten working yet (it is tricky with the discontinuities). However, I can open an issue and work on this later (as this pull request is getting bigger). 
- removing evaluate_all_at, as this was quite a rigid method (and one always had to look up the order of the output). Now you can 'evaulate' and give in a list of variables you want (although I guess one still has to look up the variable names). 

It would be helpfully if somebody else runs the tests, as a bunch of them are still failing for me. Thanks!

ian-r-rose commented on burnman/constants.py:
```diff
@@ -19,3 +19,10 @@
+Newtonian constant of gravitation (G) in m^3 kg^-1 s^-2
+"""
+
+gravitational = scipy.constants.G
```
Not sure about this name--it's an adjective for a constant value. Perhaps Big_G, or just G, since it is namespaced by constants.py?

ian-r-rose commented:
A general question: I am looking at Dziewonski and Anderson right now, and there are still differences in pressure and gravity between Table II (pages 312 and 313) and the values in prem.txt.  Do you know what is going on there? Where is prem.txt from?

ian-r-rose commented on examples/example_build_planet.py:
```diff
@@ -1,4 +1,4 @@
-# BurnMan - a lower mantle toolkit
+    # BurnMan - a lower mantle toolkit
```
Accidental indentation? That reminds me, we should update the copyright notice...

ian-r-rose commented on burnman/seismic.py:
```diff
@@ -14,38 +21,30 @@ def __init__(self):
         depth_list : array of floats
-            Array of depths [m] to evaluate seismic model at.
-            
+        Array of depths [m] to evaluate seismic model at.
```
This, and other places: the numpydoc docstring format usually has an indentation in front of the description of a parameter. Not sure whether this affects the sphinx parsing.

sannecottaar commented:
I don't know why these values are given in the paper, but not in the
version published on the IRIS website.
Gravity is from the old grav_for_prem.txt. I think I computed this back at
Cider, but I forgot the details. The pressures are from the  PREM input
file to a normal mode code. So it has become quite a jumble. I will try to
hunt a more consistent file again.

Your other points are good, thanks!


On Tue, Sep 15, 2015 at 6:08 PM, Ian Rose <notifications@github.com> wrote:

> A general question: I am looking at Dziewonski and Anderson right now, and
> there are still differences in pressure and gravity between Table II (pages
> 312 and 313) and the values in prem.txt. Do you know what is going on
> there? Where is prem.txt from?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geodynamics/burnman/pull/107#issuecomment-140465455>.
>

ian-r-rose commented:
The IRIS-PREM model sounds pretty official as well, so if you think that is more appropriate, I have no objections... maybe a note to that effect?

ian-r-rose commented on misc/ref/debye.py.out:
```diff
@@ -1 +1,2 @@
-error 7.013244e-14
+error 7.009307e-14
+time old 0.525255, time new 0.319653
```
I don't think anything should have changed here, and the error looks like machine precision, so there should be no reason to update these test results.

ian-r-rose commented on misc/ref/example_seismic.py.out:
```diff
@@ -0,0 +1,3 @@
+pressure is not defined for IASP91
+gravity is not defined for IASP91
+density is not defined for IASP91
```
When I run the script I also get warning output about pressure and gravity for STW105 and AK135

sannecottaar replied:
Yes, I put in warnings whenever it computes these values and they aren't
given. This is to warn if you put in your own, say upper mantle model, and
these calculations go totally wrong. I guess I could put in tests if it is
a whole earth model, but then again, we should maybe also add a moon model.

On Tue, Sep 15, 2015 at 7:54 PM, Ian Rose <notifications@github.com> wrote:

> In misc/ref/example_seismic.py.out
> <https://github.com/geodynamics/burnman/pull/107#discussion_r39550172>:
>
> > @@ -0,0 +1,3 @@
> > +pressure is not defined for IASP91
> > +gravity is not defined for IASP91
> > +density is not defined for IASP91
>
> When I run the script I also get warning output about pressure and gravity
> for STW105 and AK135
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geodynamics/burnman/pull/107/files#r39550172>.
>

ian-r-rose replied:
I think the warning is fine, but the test output is out of date with that
warning.

On Tue, Sep 15, 2015 at 11:58 AM, Sanne Cottaar <notifications@github.com>
wrote:

> In misc/ref/example_seismic.py.out
> <https://github.com/geodynamics/burnman/pull/107#discussion_r39550627>:
>
> > @@ -0,0 +1,3 @@
> > +pressure is not defined for IASP91
> > +gravity is not defined for IASP91
> > +density is not defined for IASP91
>
> Yes, I put in warnings whenever it computes these values and they aren't
> given. This is to warn if you put in your own, say upper mantle model, and
> these calculations go totally wrong. I guess I could put in tests if it is
> a whole earth model, but then again, we should maybe also add a moon model.
> … <#14fd25f2d1491efd_>
> On Tue, Sep 15, 2015 at 7:54 PM, Ian Rose <notifications@github.com>
> wrote: In misc/ref/example_seismic.py.out <
> https://github.com/geodynamics/burnman/pull/107#discussion_r39550172>: >
> @@ -0,0 +1,3 @@ > +pressure is not defined for IASP91 > +gravity is not
> defined for IASP91 > +density is not defined for IASP91 When I run the
> script I also get warning output about pressure and gravity for STW105 and
> AK135 — Reply to this email directly or view it on GitHub <
> https://github.com/geodynamics/burnman/pull/107/files#r39550172>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geodynamics/burnman/pull/107/files#r39550627>.
>

ian-r-rose commented on burnman/seismic.py:
```diff
@@ -170,49 +201,77 @@ def depth(self, pressure):
 
 
-class SeismicRadiusTable(Seismic1DModel):
+class SeismicTable(Seismic1DModel):
```
Any reason for the name change? Not that I object to it...

sannecottaar replied:
Hmm, might be old. I think seven months ago, I tried to make it flexible
for both radius and depth, but have since taken that out again. But mostly
we search by depth anyways, so I think this shorter name is fine.

On Tue, Sep 15, 2015 at 7:58 PM, Ian Rose <notifications@github.com> wrote:

> In burnman/seismic.py
> <https://github.com/geodynamics/burnman/pull/107#discussion_r39550681>:
>
> >          """
> >          raise ValueError, "not implemented"
> >          return -1
> >
> >
> > -class SeismicRadiusTable(Seismic1DModel):
> > +class SeismicTable(Seismic1DModel):
>
> Any reason for the name change? Not that I object to it...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geodynamics/burnman/pull/107/files#r39550681>.
>

ian-r-rose commented on burnman/seismic.py:
```diff
@@ -4,6 +4,13 @@
 import tools
+import constants
+import warnings
+import scipy.interpolate
```
This module is not currently used, until we attempt to do spline integration.

ian-r-rose commented on burnman/seismic.py:
```diff
@@ -14,38 +20,31 @@ def __init__(self):
         Parameters
         ----------
+        vars_list : array of str
+	    Variables to be evaluated. Options are depending on the model 'pressure','density','gravity','v_s','v_p','v_phi','G','K','QG','QK'
```
This was kind of difficult for me to read. Perhaps
```
Available variables depend on the seismic model, and can be chosen from ...
```

ian-r-rose commented on burnman/seismic.py:
```diff
@@ -14,38 +20,31 @@ def __init__(self):
+        
+        values=np.empty((len(vars_list),len(depth_list)))
+        for a in range(len(vars_list)):
+            values[a,:]=getattr(self,vars_list[a])(depth_list)
```
I like this change. Perhaps it would be a good idea to check whether a seismic model has this particular variable implemented? Of course, if it does not then an exception will be raised, so maybe that is enough.

sannecottaar replied:
Yeah, the attributeerror is quite clear.

ian-r-rose commented:
The output for example_compare_all_methods needs to be updated as well, though it may be difficult to do that from your end due to the difference in where warnings are piped... :(

ian-r-rose commented on examples/example_seismic.py:
```diff
@@ -41,104 +41,78 @@
-    plt.ylabel('km/s')
+    
+    ## Alternatively one is able to evaluate all the variables for a model in a single line
+    #  values = model.evaluate(vars_list,depth_list)
```
This commented out line is unnecessary.

ian-r-rose commented on examples/example_seismic.py:
```diff
@@ -41,104 +41,78 @@
+                #depths = np.arange(700e3,2800e3,100e3)
+                #3. we could also use the data points where the seismic model is specified over a depth range, this will bring out any discontinuities
+                #this is the preferred way to plot seismic discontinuities correctly
+                depths = models[m].internal_depth_list(mindepth=-1.e3, maxdepth=6372.1e3)
```
Why are we starting from a negative minimum depth? Perhaps a comment about that?

ian-r-rose commented on examples/example_seismic.py:
```diff
@@ -41,104 +41,78 @@
+    models=[burnman.seismic.PREM(), burnman.seismic.STW105(),burnman.seismic.AK135(),burnman.seismic.IASP91()]
+    colors = ['r','b','m','k']
+    # Variables to plot
+    vars=['pressure','gravity','v_p','v_s','v_phi','density']#,
```
vars() is an internal python function which you are hiding here. Another name for this would be better

tjhei replied:
also remove the comment behind it

ian-r-rose commented on examples/example_seismic.py:
```diff
@@ -41,104 +41,78 @@
+                #3. we could also use the data points where the seismic model is specified over a depth range, this will bring out any discontinuities
+                #this is the preferred way to plot seismic discontinuities correctly
+                depths = models[m].internal_depth_list(mindepth=-1.e3, maxdepth=6372.1e3)
+                #now evaluate everything at the given depths levels (using interpolation)
```
using interpolation --> using linear interpolation

ian-r-rose commented on examples/example_seismic.py:
```diff
@@ -41,104 +41,78 @@
+                #this is the preferred way to plot seismic discontinuities correctly
+                depths = models[m].internal_depth_list(mindepth=-1.e3, maxdepth=6372.1e3)
+                #now evaluate everything at the given depths levels (using interpolation)
+                try:
```
This try/except block is very economical, but maybe not clear to somebody who doesn't know what is going on.  I think some more comments are in order about what you are doing here.

tjhei commented on burnman/seismic.py:
```diff
@@ -170,49 +200,77 @@ def depth(self, pressure):
     def pressure(self, depth):
-        
-        return self._lookup(depth, self.table_pressure)
+        if len(self.table_pressure)>0:
```
you can simplify this logic to say:
```
if len(bla)==0:
  compute()
return lookup(bla)
```

tjhei commented on burnman/seismic.py:
```diff
@@ -170,49 +200,77 @@ def depth(self, pressure):
+                return self._lookup(depth, self.table_pressure)
+
+    def gravity(self, depth):
+        if len(self.table_gravity)>0:
```
same here

tjhei commented on burnman/seismic.py:
```diff
@@ -14,38 +20,31 @@ def __init__(self):
 
 
-    def internal_depth_list(self):
+    def internal_depth_list(self,mindepth=0.,maxdepth=6371.e3):
```
do we need the defaults here?

sannecottaar replied:
Why not? Now it still works in the old way where you just get the depths in the list (at least if it is smaller than Earth).

tjhei replied:
Which means you are hardcoding this for Earth. Couldn't you put in something like 1e100 for ```maxdepth```?

tjhei commented:
Looks pretty good otherwise (if tests are working)

sannecottaar commented:
I removed the gravity from the PREM file, and the values are now computed
(and are a lot closer to the values in the PREM paper). I guess that is
more honest, since I remember computing them before, and they are not from
some reference file.  The pressure values are not exact, but they are close
enough (I think).

On Tue, Sep 15, 2015 at 7:28 PM, Ian Rose <notifications@github.com> wrote:

> The IRIS-PREM model sounds pretty official as well, so if you think that
> is more appropriate, I have no objections... maybe a note to that effect?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geodynamics/burnman/pull/107#issuecomment-140491109>.
>

sannecottaar commented:
Thanks guys for all the feedback. 
Ian, would you run the tests? Indeed I have trouble dealing with the new warnings, maybe somebody can fix that right after merging this. But other tests are also failing as I don't have some text_2path or numdiff. Thanks!

ian-r-rose commented:
For what it's worth, I also have a `text_2path` related error, but I thought it was just me. I think it is due to a newer version of matplotlib not working for the `Template` backend. This is probably a bug in matplotlib.

Sanne, can you try running the test script after commenting out line 21? You will have to close a bunch of figures, but you may see the `text_2path` error disappear.

ian-r-rose commented:
Is the warning about calculating pressure and gravity still accurate? Also, I think the wording of the warning is kind of unclear regardless.

ian-r-rose commented:
Okay @sannecottaar, when this is ready, I can update the test results.

sannecottaar commented:
Updated the wording of the warning. Can we merge?

tjhei commented:
What about the max_depth default value for |internal_depth_list| in the base class? I think the default there should also be a large value instead of Earth's depth. The documentation could also be extended to explain these min/max values.

sannecottaar commented:
Sorry, had only fixed this in SeismicTable.

tjhei commented:
Looks good to me now. @ian-r-rose ?

ian-r-rose commented:
@sannecottaar, can you cherry-pick this test-results update from my repo? 
ian-r-rose@0b59eae7f2917bf7ef19cd13f479b164ec846f36

ian-r-rose commented:
Okay, looks good from my end. Someday it may be nice to revisit what is going on with the pressure and gravity profiles in PREM, but I don't think it is fruitful at the moment.
